### PR TITLE
Ui for coin freeze cltv

### PR DESCRIPTION
**Pull request by marlengit**
There are CLTV features in the protocol layer but no features in the wallet for users to use them. Currently CLTV balances will not be recognized in the wallet. So this PR includes:
1) On the Receive tab there is a new "Coin freeze" checkbox. On clicking a dialog appears with 2 new input boxes Freeze by block and Freeze by datetime. Entering freeze release values causes the Show Address dialog to display freeze details and the P2SH address which can be used for freezing.   
2) On the Send tab users can then send their coins into the freeze address using a normal send transaction. The Send confirmation displays a warning if the destination is a freeze address.  Once the coins arrive they appear as Watching Only values while they remain frozen. To unfreeze the user only has to wait until the freeze block/datetime arrives. Then the Watch Only balance shifts back to the normal Spendable balance automatically.  
3) To spend from the freeze address the coin control box will automatically display the coins when they become unfrozen. Note that lock-by-blockheight & lock-by-blocktime coins cannot be mixed into the same tx. But each kind works fine independently.
4) The Transactions tab now includes addresses and labels for txs sent to self (no more n/a in the label column). This helps viewing and searching for past txs by address/label which is very helpful if a freeze address is labelled.

There are some TODO comments that need verification. There could be limits added to prevent freezing beyond 5 or 10 years. 

BE CAREFUL TO RUN THIS ON TESTNET ONLY PENDING REVIEW.

Screenshots attached
coin-freeze-dialog1
show-coin-freeze
warning-send-freeze

### CODE REVIEW

**Comment by dgenr8** [comment]
@marlengit Are you planning to organize into a few logical commits to make review easier?

Also in case you haven't seen it https://github.com/bitcoin/bitcoin/pull/7534 .

File: src/qt/addresstablemodel.cpp [hunk 1]
```diff
@@ -85,7 +86,7 @@ class AddressTablePriv
             BOOST_FOREACH(const PAIRTYPE(CTxDestination, CAddressBookData)& item, wallet->mapAddressBook)
             {
                 const CBitcoinAddress& address = item.first;
-                bool fMine = IsMine(*wallet, address.Get());
+                bool fMine = wallet->IsMine(address.Get());
```
Comment (BlockMechanic) [inline src/qt/addresstablemodel.cpp:89]: You need to change this , as you have modified the is mine function to include a block ie CBlockindex

**Comment by gandrewstone** [comment]
I have merged this branch with the latest changes on the dev branch, on a branch "coin_freeze_cltv".  There's a lot of great work here, but we need to fix some stuff.  I hope to work on these as part of my review, and hope that you'll continue your work here by helping me out.

Things I've found so far (running on regtest):
1. when the freeze expires, the "Available" balance in the GUI is not updated.  But if you quit the GUI and restart it, the balance correctly incorporates the unfrozen money!  So I think is just a refresh or recalc thing...

2. when specifying a date, the calendar starts at Jan, 1970.  This makes is very painful to use.  We should start it at the current date (and not let it go previous).

3. In regtest, putting in a date in 1970 actually works, and gave me a block of 626400

4. Similarly, putting in a block of 1 works when the current block is 100 (greater than 1).  I understand that this is fine from an engineer's perspective, but the GUI should probably deny it, because its likely a "bad user on device" problem.
 
5. the date specifying combo box has a dash in it.  If you remove that dash, you can't type anything.
Is there a canonical date format?  If so, we should indicate what that format is

Additional features:
6. in the "Coin Freeze until Block: ", let's make the capitalization consistent with other options.  That is, "Coin freeze until block:"
7. And I'd like to also have the date in there so a user can "gut check" his choices.  so I propose:
"Coin freeze until block: 500000 (approximately Jan 1, 20XX)"
8. in the "overview" can we display the frozen coins as well (if any exist)?  I don't think we want a "frozen: " field if none exist though, since that may confuse people (but I'm ambivalent about this).

9. In the Coin Freeze dialog box: "Freezing coins means":  mean -> means,  "release" -> "moment".
and "Freeze by block" -> "Freeze until block", "Freeze by date and time" -> "Freeze until..."

**Comment by gandrewstone** [comment]
I've fixed issues 2,3 and features 6,7, 9.  I will make a PR for your review.

**Comment by marlengit** [comment]
I've been away for a while and I noticed PR #266. It will be a couple more weeks and then I will have more availability to focus on this again.

File: src/qt/addresstablemodel.cpp (hunk 1 above)
Comment (marlengit) [reply src/qt/addresstablemodel.cpp:89]: This calls an IsMine() overload within wallet.cpp::989 which calls the IsMine() within wallet_ismine.cpp passing chainActive.Tip() as the CBlockindex.

**Comment by ptschip** [comment]
@marlengit if I'm not mistaken you can close this PR now that you've opened the same on a different branch...

**Comment by marlengit** [comment]
Thanks @ptschip ... I'll close this now as the UI fixes are being reviewed via PR #266 & PR #297.

**Pull request closed without merging**